.buildkite/scripts/steps/functional/scalability_dataset_extraction.sh

```diff
@@ -28,16 +30,27 @@ for i in "${scalabilityJourneys[@]}"; do
         --without-static-resources
 done
 
-echo "--- Upload Kibana build, plugins and scalability traces to the public bucket"
-mkdir "${BUILD_ID}"
-# Archive json files with traces and upload as build artifacts
-tar -czf "${BUILD_ID}/scalability_traces.tar.gz" -C target scalability_traces
-buildkite-agent artifact upload "${BUILD_ID}/scalability_traces.tar.gz"
-# Upload Kibana build, plugins, commit sha and traces to the bucket
-download_artifact kibana-default.tar.gz ./"${BUILD_ID}"
-download_artifact kibana-default-plugins.tar.gz ./"${BUILD_ID}"
-echo "${BUILDKITE_COMMIT}" > "${BUILD_ID}/KIBANA_COMMIT_HASH"
+echo "--- Creating scalability dataset in ${OUTPUT_REL}"
+mkdir -p "${OUTPUT_DIR}"
+
+echo "--- Archiving scalability trace and uploading as build artifact"
+tar -czf "${OUTPUT_DIR}/scalability_traces.tar.gz" -C target scalability_traces
+buildkite-agent artifact upload "${OUTPUT_DIR}/scalability_traces.tar.gz"
+
+echo "--- Downloading Kibana artifacts used in tests"
+download_artifact kibana-default.tar.gz "${OUTPUT_DIR}/" --build "${KIBANA_BUILD_ID:-$BUILDKITE_BUILD_ID}"
+download_artifact kibana-default-plugins.tar.gz "${OUTPUT_DIR}/" --build "${KIBANA_BUILD_ID:-$BUILDKITE_BUILD_ID}"
```

Comment:
@dmlemeshko added back the artifact downloading, also added support for defining a KIBANA_BUILD_ID for the build which will prevent the build from being created in this BK Build, like in https://buildkite.com/elastic/kibana-single-user-performance/builds/3304#0182d5d0-980a-4d72-a1c8-1deaa82b2b5d